sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/SensorContextUtils.java

```diff
@@ -0,0 +1,269 @@
+    NewIssueLocation location = newIssue.newLocation();
+    location.on(inputFile);
+    location.message(issue.failure);
+
+    // semicolon rule
+    if (ruleKey.rule().equals("S1438")) {
+      location.at(inputFile.selectLine(issue.startPosition.line + 1));
+
+    } else if (!TypeScriptRules.FILE_LEVEL_RULES.contains(ruleKey.rule())) {
+      location.at(inputFile.newRange(
+        issue.startPosition.line + 1,
+        issue.startPosition.character,
+        issue.endPosition.line + 1,
+        issue.endPosition.character));
+    }
+
+    newIssue.at(location);
+
+    // there is not secondaryLocations for issues coming from tslint rules
+    if (issue.secondaryLocations != null) {
+      for (SecondaryLocation secondaryLocation : issue.secondaryLocations) {
+        NewIssueLocation newSecondaryLocation = newIssue.newLocation().on(inputFile);
+        setSecondaryLocation(newSecondaryLocation, secondaryLocation, inputFile);
+        newIssue.addLocation(newSecondaryLocation);
+      }
+    }
+
+    if (issue.cost != null) {
+      newIssue.gap(issue.cost);
+    }
+    LOG.info("Saving issue " + issue + "\n  as >>>> " + newIssue); // TODO Remove this
+    newIssue.save();
+  }
+
+  public static void setNodePath(File typescriptLocation, ProcessBuilder processBuilder) {
+    Map<String, String> environment = processBuilder.environment();
+    environment.put("NODE_PATH", typescriptLocation.getAbsolutePath() + File.pathSeparator + environment.getOrDefault("NODE_PATH", ""));
+  }
+
+  private static void saveIssue(SensorContext sensorContext, TypeScriptRules typeScriptRules, Issue issue, InputFile inputFile) {
+    RuleKey ruleKey = typeScriptRules.ruleKeyFromTsLintKey(issue.ruleName);
+    NewIssue newIssue = sensorContext.newIssue().forRule(ruleKey);
+    NewIssueLocation location = newIssue.newLocation();
+    location.on(inputFile);
+    location.message(issue.failure);
+
+    // semicolon rule
+    if (ruleKey.rule().equals("S1438")) {
+      location.at(inputFile.selectLine(issue.startPosition.line + 1));
+
+    } else if (!TypeScriptRules.FILE_LEVEL_RULES.contains(ruleKey.rule())) {
+      location.at(inputFile.newRange(
+        issue.startPosition.line + 1,
+        issue.startPosition.character,
+        issue.endPosition.line + 1,
+        issue.endPosition.character));
+    }
+
+    newIssue.at(location);
+
+    // there is not secondaryLocations for issues coming from tslint rules
+    if (issue.secondaryLocations != null) {
+      for (SecondaryLocation secondaryLocation : issue.secondaryLocations) {
+        NewIssueLocation newSecondaryLocation = newIssue.newLocation().on(inputFile);
+        setSecondaryLocation(newSecondaryLocation, secondaryLocation, inputFile);
+        newIssue.addLocation(newSecondaryLocation);
+      }
+    }
+
+    if (issue.cost != null) {
+      newIssue.gap(issue.cost);
+    }
+    LOG.info("Saving issue " + issue + "\n  as >>>> " + newIssue); // TODO Remove this
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this trailing comment on the previous empty line. ([squid:TrailingCommentCheck](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=sonarlint-integration-poc&issues=AWDgL5jfbBlXo5JPxg69&open=AWDgL5jfbBlXo5JPxg69)

[ISSUE_KEY]: AWDgL5jfbBlXo5JPxg69